web/client/plugins/widgetbuilder/enhancers/connection/catalogEditorConnect.js

```diff
@@ -0,0 +1,29 @@
+import { connect } from 'react-redux';
+import { compose, defaultProps } from 'recompose';
+import { changeUrl, changeType,
+    changeServiceFormat, changeMetadataTemplate, toggleAdvancedSettings, changeServiceProperty, toggleTemplate,
+    toggleThumbnail, addService, deleteService, changeTitle, changeCatalogMode } from '../../../../actions/catalog';
+import { savingSelector, newServiceSelector } from '../../../../selectors/catalog';
+import { isLocalizedLayerStylesEnabledSelector } from '../../../../selectors/localizedLayerStyles';
+
+export default compose(
+    defaultProps({id: 'dashboard-catalog-selector', buttonStyle: []}),
+    connect((state) => ({
+        saving: savingSelector(state),
+        isLocalizedLayerStylesEnabled: isLocalizedLayerStylesEnabledSelector(state),
+        service: newServiceSelector(state)
+    }), {
+        onChangeUrl: changeUrl,
+        onChangeTitle: changeTitle,
+        onChangeType: changeType,
+        onChangeServiceFormat: changeServiceFormat,
+        onChangeMetadataTemplate: changeMetadataTemplate,
+        onToggleAdvancedSettings: toggleAdvancedSettings,
+        onChangeServiceProperty: changeServiceProperty,
+        onToggleTemplate: toggleTemplate,
+        onToggleThumbnail: toggleThumbnail,
+        onAddService: addService,
+        onDeleteService: deleteService,
+        onChangeCatalogMode: changeCatalogMode
+    })
```

Comment:
We should not use the catalog state. 
We should instead use internal state of react as much as possible, and the state of the dashboard,you should store only if some change has been really applied:
 - edit and save an existing service
 - create a new service
 - delete an extisting service

So `onChangeUrl`, `onChangeTitle`, `onChangeType`, `onChangeServiceFormat`, `onChangeMetadataTemplate`, `onToggleAdvancedSettings`, `onChangeServiceProperty`, `onToggleTemplate`, `onToggleThumbnail`, `onChangeCatalogMode` should be all internal to react component.

`onAddService` (that is triggered also for update existing services and that should validate the service in react-component) and `onDeleteService` should instead be bind to the dashboard state, to a new action that UPDATE_SERVICES.
In this cases you should save in the state of the dashboard the new list of services (Copying the existing one, with local changes), and use it instead of the configured one. This list is finally saved in dashboard resource